File: WINGS_AI_Bootcamp/Haskell/Perfect_Deficient_Abundant.hs
aliquotSum :: Int -> Int
aliquotSum n = sum [x | x <- [1..n-1], n `mod` x == 0]

classify :: Int -> String
classify n
  | s == n = "Perfect"
  | s < n  = "Deficient"
  | s > n  = "Abundant"
  where s = aliquotSum n

main :: IO ()
main = print (classify 12)  -- Output: "Abundant"

File: WINGS_AI_Bootcamp/Elixir/Perfect_Deficient_Abundant.ex
defmodule NumberType do
  def aliquot_sum(n), do: sum_divisors(n, 1, 0)

  defp sum_divisors(n, i, acc) when i >= n, do: acc
  defp sum_divisors(n, i, acc) do
    if rem(n, i) == 0 do
      sum_divisors(n, i + 1, acc + i)
    else
      sum_divisors(n, i + 1, acc)
    end
  end

  def classify(n) do
    sum = aliquot_sum(n)
    cond do
      sum == n -> "Perfect"
      sum < n -> "Deficient"
      sum > n -> "Abundant"
    end
  end
end

IO.puts(NumberType.classify(12))  # Output: Abundant

